modules/telemetry/src/telemetry.cpp

```diff
@@ -0,0 +1,54 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+#include "hephaestus/telemetry/telemetry.h"
+
+#include <mutex>
+#include <vector>
+
+#include "hephaestus/telemetry/sink.h"
+
+namespace heph::telemetry {
+
+class Telemetry {
+public:
+  static void registerSink(ITelemetrySink* sink);
+
+  static void metric(const MetricEntry& log_entry);
+
+private:
+  [[nodiscard]] static auto instance() -> Telemetry&;
+
+private:
+  std::mutex sink_mutex_;
+  std::vector<ITelemetrySink*> sinks_ ABSL_GUARDED_BY(sink_mutex_);
```

Comment:
would it make sense to transfer ownership of the sinks to Telemetry and use a std::unique_ptr here?
seems dangerous to me that the sinks are defined in an unfit spot and go out of scope...

Reply:
very good question, probably you are right, as the telemetry is a singleton and global its scope is the one of the whole app if we want to avoid weird segfault we want to pass the ownership. I'll update